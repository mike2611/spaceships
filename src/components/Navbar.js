import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';
import '../App.css';
import { NavLink } from 'react-router-dom';
import planet from '../imgs/planet.png';

const Navbar = () => (
  <nav className="navBar navbar-expand-lg p-4 border border-bottom bg-white">
    <div className="container-fluid">
      <div>
        <div>
          <div className="d-flex align-items-center">
            <div className="ms-5 logo-nav">
              <span className="me-4">
                <img className="img-logo" src={planet} alt="logo" />
              </span>
              <span className="title-nav">Space Traveler&apos;s Hub</span>
            </div>
            <div>
              <NavLink
                to="/rockets"
              >
                <h2 className="links-nav">Rockets</h2>
              </NavLink>
            </div>
            <div>
              <NavLink
                to="/"
              >
                <h2 className="links-nav">Missions</h2>
              </NavLink>
            </div>
            <div className="mb-1">|</div>
            <div>
              <NavLink
                to="/myProfile"
              >
                <h2 className="links-nav">My Profile</h2>
              </NavLink>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
);

export default Navbar;
