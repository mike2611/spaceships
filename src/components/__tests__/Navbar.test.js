import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Navbar from '../Navbar';

describe('Navbar component', () => {
  test('should render Navbar component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Navbar />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
