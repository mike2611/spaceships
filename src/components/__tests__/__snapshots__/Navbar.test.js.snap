// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render Navbar component correctly 1`] = `
<nav
  className="navBar navbar-expand-lg p-4 border border-bottom bg-white"
>
  <div
    className="container-fluid"
  >
    <div>
      <div>
        <div
          className="d-flex align-items-center"
        >
          <div
            className="ms-5 logo-nav"
          >
            <span
              className="me-4"
            >
              <img
                alt="logo"
                className="img-logo"
                src="planet.png"
              />
            </span>
            <span
              className="title-nav"
            >
              Space Traveler's Hub
            </span>
          </div>
          <div>
            <NavLink
              className="nav-link"
              to="/rockets"
            >
              <h2
                className="links-nav"
              >
                Rockets
              </h2>
            </NavLink>
          </div>
          <div>
            <NavLink
              className="nav-link"
              to="/"
            >
              <h2
                className="links-nav"
              >
                Missions
              </h2>
            </NavLink>
          </div>
          <div
            className="mb-1"
          >
            |
          </div>
          <div>
            <NavLink
              className="nav-link"
              to="/myProfile"
            >
              <h2
                className="links-nav"
              >
                My Profile
              </h2>
            </NavLink>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
